Build: Add maximum compression

rasa opened the pull request:
### Purpose

Add maximum compression to .tar and .zip files.
Removed per @calmh:
~~Remove /etc files from zip file, as they are unneeded on Windows.~~
~~Fix tar builds on Windows, as some Windows users might prefer tars over zips (I do).~~
~~Add 'all' option to tar and zip targets, so all executables can be shipped, if desired.~~

### Testing

Results of `go run build.go zip` before and after (1.1% improvement):
````
-rw-r--r-- 1 ross 197121  5694763 Jun 26 13:37 syncthing-windows-amd64-v0.14.31-rc.1+5-gdb1dc998.zip
-rw-r--r-- 1 ross 197121  5629660 Jun 26 13:38 syncthing-windows-amd64-v0.14.31-rc.1+9-g7a9e7183-build-max-compression.zip
````

### Authorship

Ross Smith II (rasa) <ross@smithii.com>

calmh commented:
Hi! Thanks. In principle I'm fine with this, apart from excluding etc and extra from Windows. The extra stuff includes documentation. The etc stuff may currently be useless, but it's not guaranteed to be so in the future and it's equally useless for some other platforms we build for.

calmh commented:
Also clean up the comments on the `all` target to explain what happens with binaries instead of the current "can't build archives" thing.

rasa commented:
@calmh Thanks for reviewing, and considering this PR. I removed the exclusion of the /etc folder, and fixed the 'all' comment. Let me know if there's anything else needed. Thanks!

calmh commented on build.go:
```diff
@@ -63,8 +65,8 @@ type archiveFile struct {
 	"all": {
-		// Only valid for the "build" and "install" commands as it lacks all
-		// the archive creation stuff.
+		// Only valid for the "build", "install", "tar" and "zip" commands
```
Actually, why not add the required stuff to this target, maybe keep the later magic to add all binaries automatically, instead of reusing the syncthing target? The filenames etc will make more sense that way.

rasa replied:
Yeah, that makes more sense. Fixing now...

rasa replied:
Sorry, I started to hack on it, and I'm not sure the best way forward, sorry.

calmh commented:
Maybe scale it back to just the compression changes and add the other stuff at a later time when appropriate.

rasa commented:
OK, I backed out the build all changes. Let me know if this fits the bill.

calmh commented:
@st-jenkins retest this please

calmh commented:
@st-review merge it

build: Use maximum compression when archiving

calmh commented:
@rasa Thanks!

st-review commented:
:ok_hand: Merged as 0ca2ed7ad7aa7cd33d657e37233c3805b401c128. Thanks, @rasa!
